### PR TITLE
[ContractKit] Add secret retrieval capability to AzureKeyVaultClient

### PR DESCRIPTION
**Pull request by codyborn**
## Description

Add secret retrieval capability to AzureKeyVaultClient. This is needed for the phone number service to use BLS keys in KeyVault.

## Tested

Tested as part of this PR:
https://github.com/celo-org/celo-monorepo/pull/3508

### CODE REVIEW

**Review by tkporter — APPROVED** [review]
LGTM!

**Review by jmrossy — APPROVED** [review]
LGTM too but may be good to get signoff from @mcortesi or @gastonponti as well :)

File: packages/contractkit/src/utils/azure-key-vault-client.ts [hunk 1]
```diff
@@ -123,6 +126,14 @@ export class AzureKeyVaultClient {
     return true
   }
 
+  public async getSecret(secretName: string): Promise<string> {
+    const secret = await this.secretClient.getSecret(secretName)
+    if (typeof secret.value === 'undefined') {
```
Comment (gastonponti) [inline packages/contractkit/src/utils/azure-key-vault-client.ts]: you could check with just `if (secret.value)`
for example an empty string it's a valid one?

**Review by gastonponti — COMMENTED** [review]
All the other things looks good, but I have my doubts with that typeof === 'undefined'

File: packages/contractkit/src/utils/azure-key-vault-client.ts (hunk 1 above)
Comment (codyborn) [reply packages/contractkit/src/utils/azure-key-vault-client.ts]: Technically empty string is valid, but unlikely correct.

**Review by gastonponti — APPROVED** [review]
Great, LGTM

**Pull request merged**